test/gurka/test_route_summary.cc

```diff
@@ -205,3 +205,53 @@ TEST(TestRouteSummary, DupSummaryFix) {
 
   std::filesystem::remove_all(workdir);
 }
+
+TEST(Standalone, TripLegSummary) {
+  const std::string ascii_map = R"(
+      A---B---C---D
+    )";
+
+  const gurka::ways ways = {{"AB", {{"highway", "motorway"}}},
+                            {"BC", {{"highway", "motorway"}, {"toll", "yes"}}},
+                            {"CD", {{"route", "ferry"}}}};
+  const double gridsize = 100;
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+  const std::string workdir = "test/data/gurka_test_route_summary";
+
+  valhalla::gurka::map map = gurka::buildtiles(layout, ways, {}, {}, workdir);
+
+  valhalla::Api result0 = gurka::do_action(valhalla::Options::route, map, {"A", "B"}, "auto",
+                                           {{"/directions_type", "none"}});
+  EXPECT_TRUE(result0.trip().routes(0).legs(0).summary().has_highway());
```

Comment:
Can you also add a few EXPECT to see that the response JSON has those attributes as well? Currently only PBF response is tested. You can look at the signature of do_action, one overload accepts somewhere a string ref where it’ll put the JSON response in. Lots of other gurka tests are doing that too.

Reply:
Sure, something like this is ok? https://github.com/valhalla/valhalla/pull/4706/commits/7965887a0f3eda82a8c77b6057d68c269df4c974

Reply:
yeah perfect:) thanks!